### PR TITLE
[RECORDING PATH] fix errors and add convtranspose2d to recordings

### PR DESCRIPTION
**Pull request by johncalesp**
1. fixed some errors in parameter_generator.py
2. added convtranspose2d to recordings

### CODE REVIEW

File: tools/recording/parameter_generator.py [hunk 1]
```diff
@@ -75,10 +75,13 @@ def generate_sample(self):
...
             elif self._ops == "linear":
-                return [
-                    self.round(sample[0][0]),  # in_features
-                    self.round(sample[1][0]),  # out_features
-                ]
+                in_features = self.round(sample[0][0])
+                out_features = self.round(sample[1][0])
+                if in_features != 0 and out_features != 0:
+                    return [
+                        self.round(sample[0][0]),  # in_features
```
Comment (jimgao1) [inline tools/recording/parameter_generator.py:82]: Why not return `in_features` and `out_features` here?

File: tools/recording/record_common.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 logger = logging.getLogger(__name__)
 
+SPECIAL_SAMPLING_OPS = ['conv2d','linear']
```
Comment (jimgao1) [inline tools/recording/record_common.py:17]: Add some explanation for what this variable means

**Review by jimgao1 — CHANGES_REQUESTED** [review]

File: tools/recording/parameter_generator.py (hunk 1 above)
Comment (johncalesp) [reply tools/recording/parameter_generator.py]: changed to in_features and out_features

File: tools/recording/record_common.py (hunk 2 above)
Comment (johncalesp) [reply tools/recording/record_common.py:21]: added more comments

**Review by jimgao1 — APPROVED** [review]

**Pull request merged**